### PR TITLE
Bug: Fix Eyedropper Karma Test

### PR DESCRIPTION
**Pull request by maxyinger**
## Context
Karma test for eye dropper is currently failing on main. This fixes the existing karma test up until the point of the eyedropper having a misalignment issue at it's core. once the eyedropper is fixed this karma test should work properly.

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: assets/src/edit-story/components/colorPicker/karma/eyedropper.karma.js [hunk 1]
```diff
@@ -52,23 +52,28 @@ describe('Eyedropper', () => {
...
-    const canvas = fixture.editor.canvas.framesLayer.fullbleed;
-    await fixture.events.mouse.seq(({ down, moveRel, up }) => [
-      moveRel(image, 1, 1),
-      down(),
-      moveRel(canvas, 30, 30),
-      up(),
-    ]);
+    // Only regestering image click by clicking twice for some reason
+    await fixture.events.click(image);
+    await fixture.events.click(image);
```
Comment (barklund) [inline assets/src/edit-story/components/colorPicker/karma/eyedropper.karma.js:57]: Yeah, this is the issue we've also recorded in #7481 and that we need to fix in a general manner.

File: assets/src/edit-story/components/colorPicker/karma/eyedropper.karma.js [hunk 2]
```diff
@@ -52,23 +52,28 @@ describe('Eyedropper', () => {
...
 
     // Click the background element
-    await fixture.events.click(
-      fixture.editor.canvas.framesLayer.frames[0].node
+    const bgCanvasRect =
+      fixture.editor.canvas.framesLayer.frames[0].node.getBoundingClientRect();
+    // Make sure to click on part of bg element not covered by media element previously
+    // assed to canvas
```
Comment (barklund) [inline assets/src/edit-story/components/colorPicker/karma/eyedropper.karma.js]: 😂 😂 😂 😂 

```suggestion
    // added to canvas
```

Comment (barklund) [inline assets/src/edit-story/components/colorPicker/karma/eyedropper.karma.js]: 😂 😂 😂 😂 

```suggestion
    // added to canvas
```

File: assets/src/edit-story/components/colorPicker/karma/eyedropper.karma.js [hunk 3]
```diff
@@ -52,23 +52,28 @@ describe('Eyedropper', () => {
...
+    await fixture.events.mouse.click(
+      bgCanvasRect.left + 10,
+      bgCanvasRect.bottom - 10
```
Comment (barklund) [inline assets/src/edit-story/components/colorPicker/karma/eyedropper.karma.js]: We actually have [`clickOn`](https://github.com/google/web-stories-wp/blob/main/assets/src/karma-fixture/events.js#L314) for that same purpose:

```
await fixture.events.mouse.clickOn(fixture.editor.canvas.framesLayer.frames[0].node, 10 ,10);
```

File: assets/src/edit-story/components/colorPicker/karma/eyedropper.karma.js (hunk 2 above)
Comment (maxyinger) [reply assets/src/edit-story/components/colorPicker/karma/eyedropper.karma.js]: 🍑

File: assets/src/edit-story/components/colorPicker/karma/eyedropper.karma.js (hunk 3 above)
Comment (maxyinger) [reply assets/src/edit-story/components/colorPicker/karma/eyedropper.karma.js]: ooo very cool. thanks for sharing!

File: assets/src/edit-story/components/colorPicker/karma/eyedropper.karma.js [hunk 4]
```diff
@@ -77,6 +81,12 @@ describe('Eyedropper', () => {
     const imageOnCanvasRect = (
       await getCanvasElementWrapperById(imageOnCanvas.id)
     ).getBoundingClientRect();
+    // uncomment and view karma run to see how this isn't properly aligned
+    // await fixture.events.mouse.move(
+    //   imageOnCanvasRect.right - 2,
+    //   imageOnCanvasRect.top + 8
+    // );
+    // await fixture.events.sleep(10000);
```
Comment (maxyinger) [inline assets/src/edit-story/components/colorPicker/karma/eyedropper.karma.js:89]: @merapi @barklund if you uncomment this, and run `npm run test:karma:edit-story:watch` you can see the misalignment issue I mentioned.

Any thoughts on this? seems like an issue with the eyedropper and not the test, but maybe I'm misunderstanding something. Anyway, y'all have more context here. curious to see what you think

Comment (merapi) [reply assets/src/edit-story/components/colorPicker/karma/eyedropper.karma.js:89]: Found it - I can replicate the issue on one of my machines, will fix it today/tomorrow, right now I will work on a general fix for Moveable (double click/dnd issue).

**Comment by merapi** [comment]
The Moveable fix is already merged and the eyedropper fix will be in another PR, so I'm closing this.

**Pull request closed without merging**